src/core/protocol-authorization.ts

```diff
@@ -283,62 +261,59 @@ export class ProtocolAuthorization {
    * Verifies the actions specified in the given message matches the allowed actions in the rule set.
    * @throws {Error} if action not allowed.
    */
-  private static verifyAllowedActions(
+  private static async verifyAllowedActions(
     tenant: string,
-    author: string | undefined,
-    incomingMessageMethod: DwnMethodName,
+    incomingMessage: RecordsRead | RecordsWrite,
+    recordsWrite: RecordsWrite,
     inboundMessageRuleSet: ProtocolRuleSet,
     ancestorMessageChain: RecordsWriteMessage[],
-  ): void {
-    const inboundMessageAction = methodToAllowedActionMap[incomingMessageMethod];
+    messageStore: MessageStore,
+  ): Promise<void> {
+    const inboundMessageAction = methodToAllowedActionMap[incomingMessage.message.descriptor.method];
+    const author = incomingMessage.author;
 
     const actionRules = inboundMessageRuleSet.$actions;
-    if (actionRules === undefined) {
-      // if no action rule is defined, owner of DWN can do everything
-      if (author === tenant) {
-        return;
-      } else {
-        throw new Error(`no action rule defined for ${incomingMessageMethod}, ${author} is unauthorized`);
-      }
+    if (author === tenant) {
```

Comment:
This addresses a tangential bug that @csuwildcat and I noticed. Previously, if there were no `$actions` rules, the tenant was allowed access; or if there were rules, the rules determined the tenant's access. Now the tenant always has access.